File: Task1.py
'''
Problem 1 : "Daily Steps Tracker"
Description:
You have a list of the number of steps taken each day in a month. 
Your task is to write a program that performs the following:
Accepts the number of steps taken each day in the month as numbers separated by spaces.
Calculates the highest and lowest step counts.
Calculates the average daily step count.
Sorts the step counts in descending order.

'''


steps = input("Enter daily steps separated by spaces: ")

steps_list = list(map(int, steps.split()))

highest_steps = max(steps_list)
lowest_steps = min(steps_list)
average_steps = sum(steps_list) / len(steps_list)
sorted_steps = sorted(steps_list, reverse=True)

print("Highest Steps:", highest_steps)
print("Lowest Steps:", lowest_steps)
print("Average Steps:", round(average_steps, 2))
print("Sorted Steps (Descending):", sorted_steps)

File: Task1.cpp
#include <iostream>
#include <vector>
#include <sstream>
#include <algorithm>
#include <numeric> // For accumulate

using namespace std;

int main() {
    // Step 1: Accept input
    cout << "Enter daily steps separated by spaces: ";
    string input;
    getline(cin, input);

    // Step 2: Convert input to a vector of integers
    vector<int> steps;
    stringstream ss(input);
    int step;
    while (ss >> step) {
        steps.push_back(step);
    }

    // Step 3: Calculate required statistics
    int highest_steps = *max_element(steps.begin(), steps.end());
    int lowest_steps = *min_element(steps.begin(), steps.end());
    double average_steps = accumulate(steps.begin(), steps.end(), 0.0) / steps.size();
    sort(steps.begin(), steps.end(), greater<int>());

    // Step 4: Print the results
    cout << "Highest Steps: " << highest_steps << endl;
    cout << "Lowest Steps: " << lowest_steps << endl;
    cout << "Average Steps: " << average_steps << endl;
    cout << "Sorted Steps (Descending): ";
    for (int step : steps) {
        cout << step << " ";
    }
    cout << endl;

    return 0;
}
